### PR TITLE
fix(sends): Enforce token override when sending specific token via QR code

### PR DESCRIPTION
**Pull request by jophish**
### Description

For [ACT-1462](https://linear.app/valora/issue/ACT-1462/send-flow-qr-scanner-not-preserving-token-id-override). Updates the send flow + QR path to pass through a token override when entering via a token details screen.

### Test plan

Unit tests updated + manual tests. See video below.

https://github.com/user-attachments/assets/ee7a3aea-93de-4ab9-bb91-cd6fed629d61

### Related issues

- Fixes [ACT-1462](https://linear.app/valora/issue/ACT-1462/send-flow-qr-scanner-not-preserving-token-id-override).

### Network scalability

If a new NetworkId and/or Network are added in the future, the changes in this PR will:

- [x] Continue to work without code changes, OR trigger a compilation error (guaranteeing we find it when a new network is added)

### CODE REVIEW

File: src/send/actions.ts [hunk 1]
```diff
@@ -70,9 +71,13 @@ export type ActionTypes =
   | SendPaymentFailureAction
   | UpdateLastUsedCurrencyAction
 
-export const handleQRCodeDetected = (qrCode: QrCode): HandleQRCodeDetectedAction => ({
+export const handleQRCodeDetected = (
+  qrCode: QrCode,
+  defaultTokenIdOverride?: string
```
Comment (satish-ravi) [inline src/send/actions.ts:76]: nit: can we use object params?

File: src/send/utils.ts [hunk 2]
```diff
@@ -25,7 +25,8 @@ const TAG = 'send/utils'
 export function* handleSendPaymentData(
   data: UriData,
   isFromScan: boolean,
-  cachedRecipient?: Recipient
+  cachedRecipient?: Recipient,
+  defaultTokenIdOverride?: string
```
Comment (satish-ravi) [inline src/send/utils.ts:29]: switch to object params while we are at it?

**Review by satish-ravi — COMMENTED** [review]
Should we also pass around forceTokenid? I see it used in the qr scan for secure send, not sure if it's needed here though

**Review by jh2oman — APPROVED** [review]
A once satish's nits are addressed. I think force tokenId is mostly used in old payment flows so I think we are probably fine without adding it.

**Pull request merged**